cmd/languageserver.go

```diff
@@ -0,0 +1,59 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/sourcegraph/jsonrpc2"
+	"github.com/spf13/cobra"
+
+	"github.com/styrainc/regal/internal/lsp"
+)
+
+func init() {
+	languageServerCommand := &cobra.Command{
+		Use:   "language-server",
+		Short: "Run the Regal Language Server",
+		Long:  `Start the Regal Language Server and listen on stdin/stdout for client editor messages.`,
+
+		RunE: wrapProfiling(func(args []string) error {
+
+			ctx, cancel := context.WithCancel(context.Background())
```

Comment:
[nit] It's a little uncommon to not "just" do `defer cancel()`. I see that you deliberately call `cancel()` in the two select cases below; but is there a reason not to call it on the happy path, too? If so, `defer cancel()` should cover all our exits, I think.

Reply:
hmm, that's a good spot. I have removed.